libraries/chain/apply_context.cpp

```diff
@@ -50,6 +51,7 @@ action_trace apply_context::exec_one()
       {
          if( trx_context.can_subjectively_fail && control.is_producing_block() ) {
             control.check_contract_list( receiver );
+            control.check_action_list( act.account, act.name );
```

Comment:
should this be `control.check_action_list( receiver, act.name )` ?

The receiver and act.account will only differ if the original action is processed and requires a different receiver.  This means the root of that action processing tree will be act.account/act.name.  If that was on the blacklist it would never get to a point where they differ.  However, by allowing it to be receiver you could avoid situations where the black list is subverted by "bouncing" it off an acceptable contract which is just a relay

Reply:
If an acceptable contract action (e.g. `eosio.token::transfer`) is called a bad contract (for example the recipient `blackhat` of that transfer), then the `contract-blacklist` should be used to blacklist the `blackhat` account. The `action-blacklist` is for use when an otherwise good contract has a faulty action.

Also, if we want to prevent any `soft_fail`s from occurring, the producers would need to blacklist the `eosio::onerror` action. If we go with `receiver, act.name` rather than `act.account, act.name`, that would no longer work since the `eosio::onerror` action is first delivered immediately to the sending contract. It would not be feasible to add `receiver::onerror` to the action blacklist for every single account.

Reply:
ok, after discussion I agree this is fine.